lix/packages/sdk/src/database/applySchema.ts

```diff
@@ -37,7 +36,15 @@ export async function applySchema(args: { sqlite: SqliteDatabase }) {
     created_at TEXT DEFAULT CURRENT_TIMESTAMP NOT NULL
   ) strict;
 
-  CREATE INDEX IF NOT EXISTS idx_change_parent_id ON change (parent_id);
+  CREATE TABLE IF NOT EXISTS change_edge (
+    parent_id TEXT NOT NULL,
+    child_id TEXT NOT NULL,
+    PRIMARY KEY (parent_id, child_id),
+    FOREIGN KEY(parent_id) REFERENCES change(id),
+    FOREIGN KEY(child_id) REFERENCES change(id),
+    -- Prevent self referencing edges
+    CHECK (parent_id != child_id)
```

Comment:
I didn't know `CHECK` before. Very cool. SQLite now verifies that edges are not self-referencing. A unit test has been written as well.

Reply:
I recall we used this to validate the id of the entity type. So if you try to insert an change id (here postfixed with the c) "415f1f62-2017-4761-9ece-519c1e5e45e3-**c**" into a snapshot you can fail early